File: python/a06.py
# A06 How Many Guests?
# submission: https://atcoder.jp/contests/tessoku-book/submissions/38344037

n, q = map(int, input().split())
a = [0] + list(map(int, input().split()))

for i in range(n):
    a[i + 1] += a[i]

for j in range(q):
    l, r = map(int, input().split())
    print(a[r] - a[l - 1])

File: rust/tessoku-book/src/bin/a06.rs
// A06 How Many Guests?
// submission: https://atcoder.jp/contests/tessoku-book/submissions/38344081

use proconio::input;

fn main() {
    input!{
        n: usize,
        q: u32,
        a:[u32; n]
    }
    let mut b = vec![0; n + 1];
    for i in 0..n {
        b[i + 1] = b[i] + a[i];
    }
    for _j in 0..q {
        input!{
            l: usize,
            r: usize,
        }
        println!("{}", b[r] - b[l - 1]);
    }
}
